### PR TITLE
DOP-3895: Gatsby Preview URL

### PR DESCRIPTION
**Pull request by mmeigs**
### Ticket

DOP-3895

### Notes

Replaces the dummy url with the Gatsby Preview URL for the user.

Also, if there is a preview url, which means it is a Gatsby Preview build, this preview URL is used in the `GithubCommenter` class in place of the standard staging url.

This preview url structure is still up for debate and depends on implementation.

Note: deletes unused imports as well.

### CODE REVIEW

File: api/handlers/jobs.ts [hunk 1]
```diff
@@ -235,10 +238,8 @@ export async function snootyBuildComplete(event: APIGatewayEvent): Promise<APIGa
     await client.connect();
     const db = client.db(c.get<string>('dbName'));
     const jobRepository = new JobRepository(db, c, consoleLogger);
-    await jobRepository.updateWithCompletionStatus(jobId, null, false);
-    // Placeholder preview URL until we iron out the Gatsby Cloud site URLs.
-    // This would probably involve fetching the URLs in the db on a per project basis
-    const previewUrl = 'https://www.mongodb.com/docs/';
+    const { payload } = await jobRepository.updateWithCompletionStatus(jobId, null, false);
+    const previewUrl = getPreviewUrl();
```
Comment (rayangler) [inline api/handlers/jobs.ts:242]: Should we call this function with the `payload` object?

File: api/handlers/jobs.ts [hunk 2]
```diff
@@ -257,3 +258,15 @@ export async function snootyBuildComplete(event: APIGatewayEvent): Promise<APIGa
...
+/**
+ * Assembles Gatsby Preview URL address for Job post-build
+ * @param payload
+ * @returns string|undefined
+ */
+function getPreviewUrl(payload?: Payload): string | undefined {
+  if (!payload) return;
+  const { repoOwner, branchName, project } = payload;
+  const githubUsernameNoHyphens = repoOwner.split('-').join('').toLowerCase();
+  return `https://preview-mongodb${githubUsernameNoHyphens}.gatsbyjs.io/${project}/${branchName}/index`;
```
Comment (rayangler) [inline api/handlers/jobs.ts:271]: If we wanted to support pre-production sites for Gatsby Cloud, we should probably have a way to differentiate them by name. Is it possible to do something like check the Autobuilder's current environment, and if it's `stg`, append `stg` after `preview-mongodb${githubUsernameNoHyphens}`?

Comment (mmeigs) [reply api/handlers/jobs.ts:271]: That sounds perfect!

File: api/handlers/jobs.ts (hunk 1 above)
Comment (mmeigs) [reply api/handlers/jobs.ts:242]: Oh yes, I was testing out it being undefined. So sorry, thanks for catching!!

File: api/handlers/jobs.ts (hunk 2 above)
Comment (mmeigs) [reply api/handlers/jobs.ts]: @rayangler changed! However, I always get confused about this. Should it encompass '`dotcomstg`' as well?

Comment (rayangler) [reply api/handlers/jobs.ts]: `stg` webhooks should be reserved for GitHub push/content staging jobs while `dotcomstg` should be for test deploys, to sort of emulate the expected `prd` vs. `dotcomprd` behavior in production (but I don't think we enforce nor care too much, especially in pre-production). I suggested only `stg` under the assumption that we'd only ever use this webhook for either `prd` or `stg` data for now

**Review by rayangler — APPROVED** [review]
LGTM!

**Pull request merged**